modules/Sources/Features/Root/RootInitialization.swift

```diff
@@ -296,14 +296,16 @@ extension Root {
                     
                     let addressBookEncryptionKeys = try? walletStorage.exportAddressBookEncryptionKeys()
                     if addressBookEncryptionKeys == nil {
-                        // TODO: str4d
-                        // here you know the encryption key for the address book is missing, we need to generate one
-                        
-                        // here you have `storedWallet.seedPhrase.seedPhrase`, a seed as String
-
-                        // once the key is prepared, store it
-                        // let keys == AddressBookEncryptionKeys(key: "")
-                        // try walletStorage.importAddressBookEncryptionKeys(keys)
+                        var keys = AddressBookEncryptionKeys.empty
+                        keys.cacheFor(
+                            seed: seedBytes,
+                            account: 0,
+                            network: zcashSDKEnvironment.network
+                        )
+                        // TODO: Why are the address book keys stored in wallet storage?
+                        // Doesn't that mean they require the same authentication as the
+                        // seed phrase, instead of being accessible without biometric auth?
```

Comment:
There could be situations where you want the history and address book to be behind biometric auth. Maybe this is something we can ask when you add the first address book entry? 

If it's configurable then there should be a single config for the history and the address book, because it's expected that the history will contain addresses that need to be looked up in the address book.

Reply:
Note that, at least on Android, we have to store the address book encrypted so that it is encrypted when it is backed up. And since the key derivation is from the master seed, there are only two options: 
* require user auth to unlock the address book; or
* store the *derived* key in a place that does not require user auth (we obviously do not want to store the master seed in such a place).

In the latter case, storing the ciphertext and storing the current derived key can't be atomic, because they are in two different places that can't be written together transactionally (I think). Therefore there is necessarily a corner case where the ciphertext doesn't successfully decrypt with the stored key, e.g. when the app is killed after storing one of the ciphertext and the key, and before storing the other. In that case we have to ask for auth to rederive it from the stored salt (which *is* stored atomically with the ciphertext).

Reply:
Having written that out, I think it might be simplest to require user auth to get the master seed for the derivation now, and defer the implementation of storing the derived key if the user configures it not to require auth to another release. The argument for that is that storing the key isn't required in order for the address book encryption to work, and we would have to handle the case where the correct key isn't stored anyway.